src/Symfony/Component/HttpKernel/ControllerMetadata/Argument/ArgumentMetadata.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata\Argument;
```

Comment:
I was actually planning on creating another namespace in here: `Symfony\Component\HttpKernel\ControllerMetadata\Annotation` where I would create something similar to cache away the annotations. I will also need a sub-namespace for the Controller itself as I want to work on ControllerFactories. I don't mind changing it though.